### PR TITLE
Add Auction Types

### PR DESCRIPTION
**Pull request by RustySol**

### CODE REVIEW

File: core/types/src/response/auction.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export enum AuctionStatus {
+  CREATED,
+  STARTED,
+  COMPLETE,
+  CANCELLED
+}
```
Comment (haVincy) [inline core/types/src/response/auction.ts:6]: Should we specify enum string to help debug.
```ts
export enum AuctionStatus {
  CREATED = 'CREATED'.
  STARTED = 'STARTED',
  COMPLETE =  'COMPLETE',
  CANCELLED = 'CANCELLED'
}
```

File: core/types/src/response/auctionBid.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export enum BidStatus {
+  OPEN,
+  WITHDRAWN,
+  WON,
+  LOST
+}
```
Comment (haVincy) [inline core/types/src/response/auctionBid.ts:6]: Same here

**Review by haVincy — APPROVED** [review]

File: core/types/src/response/auction.ts (hunk 1 above)
Comment (RustySol) [reply core/types/src/response/auction.ts:6]: these are used as numbers

Comment (haVincy) [reply core/types/src/response/auction.ts:6]: You mean backend already define the numbers? Is it possible to change to use string enum for convenient debugging?

Comment (RustySol) [reply core/types/src/response/auction.ts:6]: yes the DB column is using number to indicate the statuses, just like `OrderStatus`

**Pull request merged**